Gems/Atom/RPI/Code/Source/RPI.Public/Shader/Shader.cpp

```diff
@@ -249,49 +249,53 @@ namespace AZ
             }
         }
         ///////////////////////////////////////////////////////////////////
+        
+        static bool GetPipelineLibraryPath(char* pipelineLibraryPath, size_t pipelineLibraryPathLength, const ShaderAsset& shaderAsset)
+        {
+            if (auto* fileIOBase = IO::FileIOBase::GetInstance())
+            {
+                const Data::AssetId& assetId = shaderAsset.GetId();
+                
+                Name platformName = RHI::Factory::Get().GetName();
+                Name shaderName = shaderAsset.GetName();
+
+                AZStd::string uuidString;
+                assetId.m_guid.ToString<AZStd::string>(uuidString, false, false);
 
+                char pipelineLibraryPathTemp[AZ_MAX_PATH_LEN];
+                azsnprintf(
+                    pipelineLibraryPathTemp, AZ_MAX_PATH_LEN, "@user@/Atom/PipelineStateCache/%s/%s_%s_%d.bin", platformName.GetCStr(),
+                    shaderName.GetCStr(), uuidString.data(), assetId.m_subId);
 
+                fileIOBase->ResolvePath(pipelineLibraryPathTemp, pipelineLibraryPath, pipelineLibraryPathLength);
+                return true;
+            }
+            return false;
+        }
+        
         ConstPtr<RHI::PipelineLibraryData> Shader::LoadPipelineLibrary() const
-        {
-            if (IO::FileIOBase::GetInstance())
+        { 
+            char pipelineLibraryPath[AZ_MAX_PATH_LEN] = { 0 };
+            if (GetPipelineLibraryPath(pipelineLibraryPath, AZ_MAX_PATH_LEN, *m_asset))
```

Comment:
Maybe we should save the path as a member so we don't need to to generate the same path each time the LoadPipelineLibrary or SavePipelineLibrary is call.